boxoffice/templates/boxoffice.js

```diff
@@ -210,6 +211,59 @@ $(function() {
           boxoffice.ractive.set('activeTab', boxoffice.ractive.get('tabs.selectItems.id'));
           boxoffice.ractive.scrollTop();
         },
+        applyDiscount: function(discountCoupons) {
+          //Ask server for the corresponding line_item for the discount coupon. Add one quantity of that line_item
+          $.post({
+            url: boxoffice.config.resources.kharcha.urlFor(),
+            crossDomain: true,
+            dataType: 'json',
+            headers: {'X-Requested-With': 'XMLHttpRequest'},
+            contentType: 'application/json',
+            data: JSON.stringify({
+              line_items: boxoffice.ractive.get('order.line_items').map(function(line_item) {
+                return {
+                  quantity: 1,
+                  item_id: line_item.item_id
+                };
+              }),
+              discount_coupons: discountCoupons
+            }),
+            timeout: 5000,
+            retries: 5,
+            retryInterval: 5000,
+            success: function(data) {
+              var applyDiscount;
+              var line_items = boxoffice.ractive.get('order.line_items');
+              line_items.forEach(function(line_item) {
+                if (data.line_items.hasOwnProperty(line_item.item_id)) {
+                  if(data.line_items[line_item.item_id].discounted_amount && line_item.quantity_available > 0) {
+                    applyDiscount = true;
+                    line_item.show_discountprice = true;
```

Comment:
This flag is used to show discounted price on top of base price(strikethrough), only when the ticket is not in the cart.
Once the ticket has been added to the cart, the flag is set to false.
Please go through these line
https://github.com/hasgeek/boxoffice/pull/32/files#diff-b27c18f9a3aff2cd6e17b616f8ec3486R277
https://github.com/hasgeek/boxoffice/pull/32/files#diff-b27c18f9a3aff2cd6e17b616f8ec3486R288
and UI when ticket is added to the cart
https://cloud.githubusercontent.com/assets/11267213/15697608/5bf30c98-27d9-11e6-8ff4-d1859e27e9fe.png